cpp/dolfinx/io/VTKWriter.cpp

```diff
@@ -210,6 +210,22 @@ void write_ascii_mesh(const mesh::Mesh& mesh, int cell_dim,
       throw std::runtime_error("MeshFunction of lower degree than the "
                                "topological dimension is not implemented");
     }
+
+    // Build a map from topology to geometry
+    auto cell_connectivity = mesh.topology().connectivity(tdim, 0);
+    const std::int32_t num_mesh_vertices = mesh.num_entities(0);
+    std::vector<std::int32_t> topology_to_geometry(num_mesh_vertices);
+    auto x_dofs = mesh.geometry().dofmap();
+    for (int j = 0; j < mesh.num_entities(tdim); ++j)
```

Comment:
Avoid `num_entities`. It will disappear soon. In the case you can use `cell_connectivity.num_nodes()`.